trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/ACSLHandler.java

```diff
@@ -438,69 +438,72 @@ private static int getCASTBinaryExprOperator(
 	private ExpressionResult dispatchSwitch(final IDispatcher main,
 			final de.uni_freiburg.informatik.ultimate.model.acsl.ast.Expression node, final ILocation loc) {
 		final ExpressionResult expr = (ExpressionResult) main.dispatch(node, main.getAcslHook());
+		if (expr.getLrValue() instanceof final HeapLValue heapValue) {
+			final CType type = CEnum.replaceEnumWithInt(expr.getLrValue().getCType().getUnderlyingType());
+			// Perform a custom switch to RValue without any read calls
+			// This allows us to use also dereferences inside ACSL expressions that have to be side-effect-free
+			// (e.g., loop invariant or contracts)

```

Comment:
But this will now also be used for ACSL asserts, right?

Reply:
Yes, it will be used for any ACSL expression.